packages/content/src/Application/PropertyResolver/Resolver/SmartContentPropertyResolver.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Content\Application\PropertyResolver\Resolver;
+
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\OptionMetadata;
+use Sulu\Bundle\AdminBundle\SmartContent\SmartContentProviderInterface;
+use Sulu\Bundle\AudienceTargetingBundle\TargetGroup\TargetGroupStoreInterface;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Sulu\Component\Webspace\Segment;
+use Sulu\Component\Webspace\Webspace;
+use Sulu\Content\Application\ContentResolver\Value\ContentView;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @phpstan-import-type SmartContentBaseFilters from SmartContentProviderInterface
+ */
+class SmartContentPropertyResolver implements PropertyResolverInterface
+{
+    public function __construct(
+        private RequestStack $requestStack,
+        private RequestAnalyzerInterface $requestAnalyzer,
+        private ?TargetGroupStoreInterface $targetGroupStore,
+    ) {
+    }
+
+    /**
+     * @param array{
+     *     categories?: int[],
+     *     tags?: string[],
+     *     categoryOperator?: 'AND'|'OR',
+     *     tagOperator?: 'AND'|'OR',
+     *     sortBy?: string,
+     *     sortMethod?: 'ASC'|'DESC',
+     *     limitResult?: int|null,
+     *     dataSource?: string|null,
+     *     types?: string[]|null,
+     *     presentAs?: string|null,
+     *     includeSubFolders?: bool|null,
+     *     excludeDuplicates?: bool,
+     *     audienceTargeting?: bool
+     * } $data
+     * @param array<string, mixed> $params
+     */
+    public function resolve(mixed $data, string $locale, array $params = []): ContentView
+    {
+        if (!\is_array($data)) { // @phpstan-ignore function.alreadyNarrowedType
+            return ContentView::create($data, $params);
```

Comment:
If the expected type is not correct a property resolver should always return `null` for single values or empty list `[]` for multi values.


```suggestion
            return ContentView::create([], $params);
```